lib/nrf_modem_lib/trace_backends/rtt/rtt.c

```diff
@@ -72,7 +91,7 @@ struct nrf_modem_lib_trace_backend trace_backend = {
 IF_DISABLED(CONFIG_UNITY, (static))
 int nrf_modem_lib_trace_rtt_channel_alloc(void)
 {
-	const int segger_rtt_mode = SEGGER_RTT_MODE_BLOCK_IF_FIFO_FULL;
+	const int segger_rtt_mode = SEGGER_RTT_MODE_NO_BLOCK_TRIM;
```

Comment:
We are subtracting the remaining bytes on line 68. The index (`idx`) is calculated based on that, so that `&buf[idx]` points to the remaining data and `transfer_len` is the RTT buffer size or remaining length, dependent on what is less.

Reply:
I see, thanks for explaining.